src/Cache.php

```diff
@@ -88,7 +88,7 @@ public function getHash(Data $Data, $Marker = "")
     public function writeToCache($ID, Image $pChartObject)
     {
         /* Compute the paths */
-        $TemporaryFile = $this->CacheFolder . "/tmp_" . rand(0, 1000) . ".png";
+        $TemporaryFile = tempnam($this->CacheFolder, "tmp_");
```

Comment:
Hello, thank you for the PR. Perhaps you could still add the `.png` file extension, to make it easier to open it as an image?

Reply:
Oh yes, good plan. Done.

Reply:
Actually, reverted back. You can not add file extension after tempnam generated files (you can only add a prefix).
As well as generating a random file and tempnam also creates the file in the temp folder specified, therefore you can not simply append `. “.png”` after the string.

Reply:
If any help, we have been running with the patch suggested in this PR for over a year and never seen the cache error since, so it is proven to work.